changes/22941-recreate-deleted-idevice-from-mdm

```diff
@@ -0,0 +1 @@
+* Fixed a bug where BYOD iDevices deleted in Fleet but still enrolled in MDM were not re-created on the next MDM checkin.
```

Comment:
(not mentioning ABM/ADE devices since it was already working, I just added a test to cover this scenario)